### PR TITLE
fix: SentrySwiftUI Carthage build import

### PR DESCRIPTION
**Pull request by nekrich**
## :scroll: Description
tl;dr; Fixed SentrySwiftUI Carthage build import.

Since there is no modulemap for the SentrySwiftUI declaring exported submodules:
1. For Carthage build make `@_implementationOnly import SentryInternal`.
2. Import SentrySwiftUI xcframework Samples/Carthage-Validation/XCFramework project to check imports.

## :bulb: Motivation and Context

Can't import prebuild SentrySwiftUI.xcframework. Got a `No such module 'SentryInternal'` error.
<details>
<summary>🖼️ Screenshot</summary>
image
</details>
At first I thought it's library evolution again, but it was broken imports inside a private swift interface file.

## :green_heart: How did you test it?
1. Run `make build-xcframework` to create xcframework.
2. Import SentrySwiftUI.xcframework into a Samples/Carthage-Validation/XCFramework project.
3. Build Samples/Carthage-Validation/XCFramework project.

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by brustolin** [comment]
Hello @nekrich, thanks for the help with this PR!!

**Review by brustolin — APPROVED** [review]

File: Sources/SentrySwiftUI/SentryTracedView.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Foundation
 import Sentry
-#if SWIFT_PACKAGE || CARTHAGE
-import SentryInternal
+#if CARTHAGE || SWIFT_PACKAGE
```
Comment (nekrich) [inline Sources/SentrySwiftUI/SentryTracedView.swift:3]: @brustolin `import SentryInternal` will work with SPM, since there is additional lib target `SentryInternal`: https://github.com/getsentry/sentry-cocoa/blob/1ee3d54a9529f3d2343600e8af1442c00f42b3cf/Package.swift#L31-L37

Comment (brustolin) [reply Sources/SentrySwiftUI/SentryTracedView.swift:3]: It works. But we dont need to expose it, so using `@_implementationOnly` its fine
